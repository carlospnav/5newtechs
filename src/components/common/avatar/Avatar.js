import styled from 'styled-components';
import { avatarProps } from 'theme/utils';


/* Avatar Component - Renders a small or large
  avatar image based on the small prop passed to it.
*/
const Img = styled.img`
  object-fit: contain;
`

export const BigImg = styled(Img).attrs({
  src: "/static/avatar/icn-avatar-big.svg",
  width: "300px",
  height: "283px"
})`
`

export const SmallImg = styled(Img).attrs({
  src: "/static/avatar/icn-avatar-small.svg",
  width: "140px",
  height: "132.6px"
})`
`

export default function({ small = false }) {
  return (
    small ? <SmallImg /> : <BigImg />
  )
}
