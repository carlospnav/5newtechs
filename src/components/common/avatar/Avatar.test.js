import { shallow } from "enzyme"
import Avatar, { SmallImg, BigImg } from "./Avatar"
import "jest-styled-components"

describe("Avatar", () =>{
  let wrapper;    

  const avatar = () => {
    wrapper = shallow(<Avatar small={true}/>)
  }

  beforeEach(avatar)
  afterEach(() => wrapper = undefined)

  it("renders a small Avatar if 'small' props is passed as true.", () => {
    expect(wrapper.find(SmallImg)).toHaveLength(1)
    expect(wrapper.find(SmallImg).dive().dive()).toMatchSnapshot()
  })

  it("renders a small Avatar if 'small' props is passed as true.", () => {
    wrapper.setProps({small: false})
    expect(wrapper.find(BigImg)).toHaveLength(1)
    expect(wrapper.find(BigImg).dive().dive()).toMatchSnapshot()
  })
})
