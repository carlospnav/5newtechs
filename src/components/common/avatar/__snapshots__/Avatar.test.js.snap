// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders a small Avatar if 'small' props is passed as true. 1`] = `
.c0 {
  object-fit: contain;
}

<img
  className="c0"
  height="132.6px"
  src="/static/avatar/icn-avatar-small.svg"
  width="140px"
/>
`;

exports[`Avatar renders a small Avatar if 'small' props is passed as true. 2`] = `
.c0 {
  object-fit: contain;
}

<img
  className="c0"
  height="283px"
  src="/static/avatar/icn-avatar-big.svg"
  width="300px"
/>
`;
